socrata/resource.py

```diff
@@ -90,8 +90,18 @@ def from_uri(cls, auth, uri):
 
     def _clone(self, res):
         return self.__class__(self.auth, res, self.parent)
+    
+    def _handle_non_standard_responses(self, response):
+        resp_non_standard = 'resource' not in response
+        if not resp_non_standard: return response
+        modded_resp = {}
+        modded_resp['resource'] = response
+        modded_resp['links'] = response.get('links', {})
+        return modded_resp
+
 
     def _on_response(self, response):
+        response = self._handle_non_standard_responses(response)
```

Comment:
Is this codepath hit if we get a non-2xx response?  If so, that should still explode here.

Reply:
Things appear to be set up so that any non 2xx responses result in exceptions: https://github.com/socrata/socrata-py/blob/main/socrata/http.py#L61